### PR TITLE
feature/opensearch

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As an Infra Engineer I would like the option of being able to use OpenSearch instead of ElasticSearch as my vector database

## Guidance to review

Note that this is POC only, and this is made clear in the README

This can be deployed and tested to dev

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: infrastructure/aws/data.tf [hunk 1]
```diff
@@ -49,8 +49,7 @@ locals {
   }
 
   django_app_secrets = {
-    "ELASTIC__API_KEY" : var.elastic_api_key,
-    "ELASTIC__CLOUD_ID" : var.cloud_id,
+    "ELASTIC__COLLECTION_ENDPOINT": module.opensearch.collection_enpdoint,
```
Comment (gecBurton) [inline infrastructure/aws/data.tf:52]: v nasty abuse of terminology here, suggestions welcome

File: redbox-core/redbox/models/settings.py [hunk 2]
```diff
@@ -131,6 +141,25 @@ def elasticsearch_client(self) -> Elasticsearch:
...
+            auth = AWS4Auth(
+                credentials.access_key,
+                credentials.secret_key,
+                self.aws_region,
+                "aoss",
+                session_token=credentials.token,
+            )
+            client = OpenSearch(
+                hosts=[{"host": self.collection_enpdoint, "port": 443}],
+                http_auth=auth,
```
Comment (gecBurton) [inline redbox-core/redbox/models/settings.py:156]: not sure this is needed from within VPC, still investigating

File: infrastructure/aws/iam.tf [hunk 3]
```diff
@@ -46,6 +46,13 @@ data "aws_iam_policy_document" "ecs_exec_role_policy" {
     ]
   }
 
+  # Add this for OpenSearchServerless access
+  # statement {
+  #  effect = "Allow"
+  #  actions = ["aoss:*"]
+  #  resources = ["*"] restrict this
```
Comment (gecBurton) [inline infrastructure/aws/iam.tf:53]: depends upon https://technologyprogramme.atlassian.net/browse/EN-658

File: README.md [hunk 4]
```diff
@@ -173,3 +173,11 @@ Follow the instructions here https://django-q2.readthedocs.io/en/master/schedule
 3. func = `django.core.management.call_command`
 4. args = `"delete_expired_data"`
 5. save
+
+
+## Vector databases
+
+We are currently using ElasticSearch as our vector database. 
+
+We have also successfully deployed Redbox to OpenSearch Serverless but this support shouldbe considered experimental
```
Comment (252afh) [inline README.md:182]: space in `shouldbe`

**Review by 252afh — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (gecBurton) [inline README.md:182]:
```suggestion
We have also successfully deployed Redbox to OpenSearch Serverless but this support should be considered experimental 
```

**Pull request merged**